exporter/clickhousestsexporter/exporter_traces.go

```diff
@@ -23,11 +25,13 @@ type tracesExporter struct {
 	insertSQL        string
 	resourceExporter *resourcesExporter
 
-	logger *zap.Logger
-	cfg    *Config
+	logger      *zap.Logger
+	cfg         *Config
+	sizer       ptrace.Sizer
+	tracesBytes metric.Int64Counter
 }
 
-func newTracesExporter(logger *zap.Logger, cfg *Config) (*tracesExporter, error) {
+func newTracesExporter(logger *zap.Logger, set component.TelemetrySettings, cfg *Config) (*tracesExporter, error) {
```

Comment:
the exporter is quite late in the pipeline - so the metric will not precisely reflect how much data entered the system.  It may be good enough for now, but maybe we should have a receiver handling the accounting?

Reply:
I can move the code to a new receiver. So perhaps if we prefer I do it already ...